charts/openfaas-cloud/templates/edge-router-dep.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.edgeRouter.name }}
+  labels:
+    app.kubernetes.io/name: openfaas-cloud
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    app.kubernetes.io/component: {{ .Values.edgeRouter.name }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ .Values.edgeRouter.name }}
+  template:
+    metadata:
+      annotations:
+        prometheus.io.scrape: {{ .Values.global.prometheusScrape | quote }}
+      labels:
+        app: {{ .Values.edgeRouter.name }}
+    spec:
+      containers:
+        - name: {{ .Values.edgeRouter.name }}
+          image: {{ .Values.edgeRouter.container }}:{{ .Values.edgeRouter.tag }}
+          imagePullPolicy: Always
+          livenessProbe:
```

Comment:
As a note - eventually the `http` checks will cause issues for Istio users. I know the YAMLs don't cover this yet, but in the future we could add exec/http selection via the approach in faas-netes https://github.com/openfaas/faas-netes/blob/master/chart/openfaas/values.yaml#L7

Reply:
I have added the same functionality as faas-netes's chart, with an option to switch to an exec check